src/gui/search/searchtab.h

```diff
@@ -31,21 +31,25 @@
 #ifndef SEARCHTAB_H
 #define SEARCHTAB_H
 
-#include <QWidget>
+#include "searchsortmodel.h"
+#include "ui_searchtab.h"
 
+#define ENGINE_URL_COLUMN 4
+#define URL_COLUMN 5
+
+QT_BEGIN_NAMESPACE
 class QLabel;
 class QTreeView;
 class QHeaderView;
 class QStandardItemModel;
-class QSortFilterProxyModel;
-class QModelIndex;
 class QVBoxLayout;
+QT_END_NAMESPACE
 
 class SearchSortModel;
 class SearchListDelegate;
 class SearchWidget;
 
-class SearchTab: public QWidget
+class SearchTab: public QWidget, private Ui::SearchTab
```

Comment:
It is preferable to use aggregation, not private inheritance from Ui classes. This eliminates naming conflicts and some of the other shortcomings.